lib/puppet/provider/package/pip.rb

```diff
@@ -87,9 +87,9 @@ def self.instances(target_command = nil)
   end
 
   # Parse lines of output from `pip freeze`, which are structured as:
-  # _package_==_version_
+  # _package_==_version_ or _package_===_version_
   def self.parse(line)
-    if line.chomp =~ /^([^=]+)==([^=]+)$/
+    if line.chomp =~ /^([^=]+)===?([^=]+)$/
```

Comment:
if the pip returns something like `pytz===2012f`, then we'll set `ensure` to that value. Does puppet ever try to interpret that as a semantic version string? It don't think it'll be a problem, but wanted to ask. For example, https://github.com/puppetlabs/puppet/blob/c2ccdd4efe8ffc4a721ab47c88a834158a37251e/lib/puppet/type/package.rb#L128-L140 and https://github.com/puppetlabs/puppet/blob/c2ccdd4efe8ffc4a721ab47c88a834158a37251e/lib/puppet/type/package.rb#L195-L199 both accept arbitrary version strings.

Reply:
hmm, are you asking in relation to this specific PR or just in general? if puppet is trying to interpret `2012f` as a semantic version higher up in some other code, i would imagine it would be broken in the wild currently because there are plenty of python packages (and i'd imagine other provider package types) that don't conform to semantic versioning.

Reply:
> if puppet is trying to interpret 2012f as a semantic version higher up in some other code, i would imagine it would be broken in the wild currently 

yeah that's a good point. 👍